### ISSUE TITLE
When I set shuffle=False,num_workers>0 in NodeDataLoader,  I can not obtain ordered samples

### ISSUE THREAD

**Issue by zzzzzero** [dmlc/dgl#5068]
When I set shuffle=False,num_workers=0 in NodeDataLoader,  I can obtain ordered samples.However,when I set shuffle=False,num_workers>0 in NodeDataLoader,   I can not  obtain ordered samples.
Here is the example:
```
import dgl
import numpy as np
sampler = dgl.dataloading.MultiLayerFullNeighborSampler(2)
import scipy.sparse as sp
spmat = sp.rand(10000, 10000, density=0.05) #
g=dgl.from_scipy(spmat)  
dataloader = dgl.dataloading.NodeDataLoader(g, np.arange(10000), sampler,batch_size=1024, shuffle=False, drop_last=False, num_workers=0)
for input_nodes, output_nodes, blocks in dataloader:
  dst_idx=blocks[-1].dstdata[dgl.NID].numpy()
  print(dst_idx[:5])
```
[0 1 2 3 4]
[1024 1025 1026 1027 1028]
[2048 2049 2050 2051 2052]
[3072 3073 3074 3075 3076]
[4096 4097 4098 4099 4100]
[5120 5121 5122 5123 5124]
[6144 6145 6146 6147 6148]
[7168 7169 7170 7171 7172]
[8192 8193 8194 8195 8196]
[9216 9217 9218 9219 9220]

```
import dgl
import numpy as np
sampler = dgl.dataloading.MultiLayerFullNeighborSampler(2)
import scipy.sparse as sp
spmat = sp.rand(10000, 10000, density=0.05) #
g=dgl.from_scipy(spmat)  
dataloader = dgl.dataloading.NodeDataLoader(g, np.arange(10000), sampler,batch_size=1024, shuffle=False, drop_last=False, num_workers=4)
for input_nodes, output_nodes, blocks in dataloader:
  dst_idx=blocks[-1].dstdata[dgl.NID].numpy()
  print(dst_idx[:5])
```
[0 1 2 3 4]
[3072 3073 3074 3075 3076]
[6144 6145 6146 6147 6148]
[8192 8193 8194 8195 8196]
[1024 1025 1026 1027 1028]
[4096 4097 4098 4099 4100]
[7168 7169 7170 7171 7172]
[9216 9217 9218 9219 9220]
[2048 2049 2050 2051 2052]
[5120 5121 5122 5123 5124]

You can reproduce this bug in colab , and this bug doesn't occur in dgl==0.5.2:
https://colab.research.google.com/drive/1iq5zcSLsrMp5WW6-Jh_lxsVnaLuRydqu#scrollTo=U0qqtSiey4FG

**Comment by yaox12** [comment]
This is caused by the incorrect tensor division. For example, result of the second run is:
```
woker 0: [0 1 2 3 4]
woker 1: [3072 3073 3074 3075 3076]
woker 2: [6144 6145 6146 6147 6148]
woker 3: [8192 8193 8194 8195 8196]
woker 0: [1024 1025 1026 1027 1028]
woker 1: [4096 4097 4098 4099 4100]
woker 2: [7168 7169 7170 7171 7172]
woker 3: [9216 9217 9218 9219 9220]
woker 0: [2048 2049 2050 2051 2052]
woker 1: [5120 5121 5122 5123 5124]
``` 
That's, worker 0 holds [0...3071], but it should be [0...1023] + [4096...5119] + [8192...9215].

**Comment by frozenbugs** [comment]
[Action Item]: verify this is fixed in graphbolt.

**Comment by frozenbugs** [comment]
Verified that this is not reproducible in graphbolt, closing this issue since dataloader function will be taken over by graphbolt soon.
